src/ShopApp.js

```diff
@@ -1,32 +1,50 @@
-import React from "react";
+import React, { useState } from "react";
 import {
   TopBar,
   ShopAppWrapper,
   HeaderTitle,
   HeaderMenu,
   SearchInput,
 } from "./ShopApp.styled";
-import { BrowserRouter as Router, Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import carIcon from "./utils/assets/shopping-car.json";
 import searchIcon from "./utils/assets/search-icon.json";
 import Icon from "./utils/modules/Icon";
 import { Footer } from "./components/outside/footer/Footer";
 import { AppRouter } from "./router/AppRouter";
 
 export const ShopApp = () => {
+  const [searchValue, setSearchValue] = useState("");
+  const history = useNavigate();
+
+  const goToSearch = () => {
+    history(`/search/${searchValue}`);
+  };
+
+  const handleEnter = (e) => {
+    if (e.key === "Enter") {
+      history(`/search/${searchValue}`);
+    }
+  };
+
   return (
-    <Router>
-      <ShopAppWrapper>
-        <div className="Header">
-          <TopBar>
-            <span>The Best Ecommerce Store</span>
-          </TopBar>
-          <HeaderTitle>
-            <Link to="/">Muebles Troncoso</Link>
-          </HeaderTitle>
-          <HeaderMenu>
-            <div className="searchSection">
-              <SearchInput placeholder="Search ...." />
+    <ShopAppWrapper>
+      <div className="Header">
+        <TopBar>
+          <span>The Best Ecommerce Store</span>
+        </TopBar>
+        <HeaderTitle>
+          <Link to="/">Muebles Troncoso</Link>
+        </HeaderTitle>
+        <HeaderMenu>
```

Comment:
In my opinion it would be better to move everything inside this `HeaderMenu` to a component on its own an just call it here, you already have your `footer` like that
```suggestion
        <Header />
```